File: src/python/laboneq/core/types/enums/section_timing_mode.py
# Copyright 2022 Zurich Instruments AG
# SPDX-License-Identifier: Apache-2.0

from enum import Enum


class SectionTimingMode(Enum):
    RELAXED = "relaxed"
    """Rounding to the hardware timing grid is allowed (default)."""

    STRICT = "strict"
    """Rounding to the hardware timing grid is rejected if the rounding distance
    exceeds 1/1000 of the grid step. This catches accidental off-grid values while
    tolerating floating-point representation noise."""

File: src/rust/laboneq-dsl/src/types/section_timing_mode.rs
// Copyright 2026 Zurich Instruments AG
// SPDX-License-Identifier: Apache-2.0

#[derive(Debug, Clone, PartialEq, Eq, Copy, Default)]
pub enum SectionTimingMode {
    #[default]
    Relaxed,
    Strict,
}
